### PR TITLE
Add RX input volume control + live input-level indicator (#356)

### LINKED ISSUE #356: Add input volume control and live input-level indicator (too low / just right / too high)
Feature request: add an input-volume (input gain) control and a live input-level indicator that shows whether the received audio is too low, too high (clipping/overload), or in a healthy range.

Problem
- There is no in-app control for RX/input gain. Operators using line-in, USB audio, or the phone mic cannot adjust the app's input level. Weak input levels produce poor decodes; excessive levels cause clipping and decode failures.

Proposal
- Add a persistent input-volume (gain) control and a live input-level indicator (meter) with clear visual feedback: "Too low", "Just right", "Too high".

User-facing behavior
- Settings → Audio: add an "Input volume" slider (0–200% or 0.0–2.0 gain) with a numeric percent/decibel readout and a default of 100% (1.0x).
- Persist the setting in the config DB under a key like "inputVolume" or "inputGain" so it survives restarts.
- Live meter: show a small input-level meter near the waterfall or RX strip (or in the main status bar) that updates in real time during receive. The meter should present both a numeric level (dBFS or percent) and a color-coded status:
  - Too low (blue or red) when below a configurable lower threshold
  - Just right (green) when within the target window
  - Too high (yellow/red) when approaching clipping or when sample peak/short-term RMS exceeds a high threshold
- Provide optional audible/visual clipping indicator (brief flash or icon) when the input path clips.
- When the user changes the input-volume slider, apply the gain to incoming samples before resampling/decoding so the decoder sees the adjusted level immediately. Persist changes when the user dismisses the settings dialog (or instantly if the UI already applies live changes).

Implementation notes (suggested)
- Android path: in MicRecorder.java, multiply incoming float samples by the configured input gain before pushing into the resampler / SlotAccumulator. Respect unprocessed/voice recognition selection; do not re-enable OEM AGC.
- Desktop path: in desktop/src-tauri/src/audio/input.rs (or input processing), apply the same gain multiply to captured samples before resampling/accumulation.
- Config persistence: use existing DatabaseOpr.writeConfig / DatabaseOpr.get_config conventions (DatabaseOpr.writeConfig("inputVolume", value, null) and DatabaseOpr.get_config("inputVolume")). Mirror other audio config keys already in use (e.g., audioInputDevice, audioSampleRate).
- UI: RadioAudioSettings.kt / ConfigFragment.java to add the slider and picker; TxStrip/receive UI to host the live meter. Use GeneralVariables or MainViewModel to expose mutable live input level for the UI to observe.
- Metering: compute peak and short-term RMS on a small sliding window; present value in percent or dBFS. Choose sensible default thresholds (e.g., "Too low" < 25%, "Just right" 25–75%, "Too high" > 75%) but make thresholds configurable later if needed.
- Tests: unit test applyInputGain function; integration test that gain changes affect decode SNR; manual verification with USB audio interface recommended.

Compatibility and defaults
- Default input gain: 100% (1.0) — unchanged behavior.
- Feature should not enable any OEM AGC/processing; keep chooseAudioSource semantics (UNPROCESSED fallback).

Why this helps
- Allows users to tune receive level from the app when hardware gain is inconvenient or unavailable.
- Prevents clipping and improves decode rates by keeping audio in a healthy range.

Files/areas to change (non-exhaustive)
- Android: ft8af/app/src/main/java/com/k1af/ft8af/wave/MicRecorder.java, ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/settings/RadioAudioSettings.kt, ft8af/app/src/main/java/com/k1af/ft8af/GeneralVariables.java, ft8af/app/src/main/java/com/k1af/ft8af/database/DatabaseOpr.java
- Desktop: desktop/src-tauri/src/audio/input.rs, desktop/src-tauri/src/engine.rs
- UI: main receive screen (add meter), settings screen (input gain slider)

Labels: enhancement

### PR DESCRIPTION
**Pull request by patrickrb**
Fixes #356

## What

Adds an in-app RX input gain control and a live input-level indicator so operators on line-in / USB audio / the phone mic can dial the app's receive level into a healthy range without touching hardware gain.

## Gain path

- New plain-Java `InputAudioLevel.applyGain()` multiplies incoming float samples in place in `HamRecorder.doOnWaveDataReceived` — the **single choke point** every RX audio path funnels through: the built-in mic `AudioRecord` loop and the direct-USB capture callback (both via `MicRecorder`'s data listener), plus the CAT-audio / ICOM-WiFi / Flex / Xiegu network connectors (which call `doOnWaveDataReceived` directly). The decoder, slot accumulator, waterfall FFT, and the level meter all see the same post-gain samples.
- Unity gain (100%) is an **exact no-op** — the default leaves the audio path bit-identical to previous releases.
- No OEM AGC/processing is enabled; `chooseAudioSource` (UNPROCESSED / VOICE_RECOGNITION) semantics are untouched.

## Settings UI

- Settings → Radio & Audio → Audio: new **Input Volume** row with a slider dialog, range **0–200%** (100% = 1.0x default), big numeric percent readout, ±5% step buttons. Changes apply live (the very next audio buffer uses the new gain); the value is persisted on dismiss.
- The existing TX volume slider dialog was generalized (`VolumeSliderDialog` with title/advice/max params) rather than duplicated.

## Persistence

- `DatabaseOpr.writeConfig("inputVolume", <percent>)` on dialog dismiss; restored at startup by the existing config loader into `GeneralVariables.inputGainPercent` (missing key → 100%). Included automatically in the new settings backup/export (#357) since that serializes the whole config table.

## Live meter

- `InputAudioLevel` accumulates peak + RMS over a 250 ms (3000-sample @ 12 kHz) window of the post-gain samples and publishes one snapshot per window via `GeneralVariables.mutableInputLevel` (LiveData), keeping main-thread traffic bounded regardless of capture buffer size.
- `InputLevelIndicator` in the waterfall screen's bottom strip renders an RMS-filled bar with a peak tick, a percent readout of the window peak, and a color-coded status word.

## Thresholds (on window peak, 1.0 = full scale)

| Status | Range | Color |
|---|---|---|
| Too low | < 25% | blue |
| Just right | 25–75% (inclusive) | green |
| Too high | > 75% | yellow |
| Clipping | ≥ 99% | red |

## Strings

New strings (`settings_input_volume*`, `input_level_*`) added to `strings_compose.xml` in all 16 locales (en, ar, cs, es, fr, in, it, ja, ko, nl, pl, ru, tr, uk, zh-rCN, zh-rTW).

## Tests

Decision logic is extracted into plain testable units per repo convention:

- `InputAudioLevelTest` (Java, 13 tests): gain multiply, unity-gain exactness, length clamping, null safety, windowed peak/RMS emission, window reset, multi-window buffers.
- `InputLevelIndicatorTest` (Kotlin, 17 tests): peak→status classification incl. boundary values (25%/75% inclusive, 99% clip), fraction/percent clamping, status→color mapping.

`gradlew testDebugUnitTest`: **1198 tests, 0 failures, 0 errors**. `assembleDebug` builds clean.

## Manual verification advised

The direct-libusb USB-audio capture path and the network-audio paths were verified by code inspection only — a quick on-air check with a USB audio interface (meter responsive, gain audibly affecting decode levels, no regression at 100%) is recommended before merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ft8af/app/src/main/java/com/k1af/ft8af/database/DatabaseOpr.java [hunk 1]
```diff
@@ -2573,6 +2573,10 @@ protected Void doInBackground(Void... voids) {
                     GeneralVariables.volumePercent = result.equals("") ? 1.0f : Float.parseFloat(result) / 100f;
                     GeneralVariables.mutableVolumePercent.postValue(GeneralVariables.volumePercent);
                 }
+                if (name.equalsIgnoreCase("inputVolume")) {//RX input gain (percent, 100 = unity)
+                    GeneralVariables.inputGainPercent =
+                            result.equals("") ? 1.0f : Float.parseFloat(result) / 100f;
+                }
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8af/app/src/main/java/com/k1af/ft8af/database/DatabaseOpr.java:2583]: Valid — fixed in 6f50890. Parsing now goes through a new testable helper, `InputAudioLevel.parseGainPercent(String)`: empty/non-numeric/NaN input falls back to unity gain (the default), and numeric input is clamped to [0, `MAX_GAIN`] (2.0x = the slider's 200% cap), so a corrupted or hand-edited import (#382) can neither crash startup nor push metering/decoding out of range. Note the neighboring `volumeValue`/`icomPort` keys are equally unguarded pre-existing code — left as-is per scope, but this PR's key is now safe. Covered by 7 new unit tests in `InputAudioLevelTest` (garbage, >200%, negative, NaN, Infinity, whitespace, empty/null).

File: ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/settings/RadioAudioSettings.kt [hunk 2]
```diff
@@ -778,13 +816,17 @@ private fun SerialPortPickerDialog(
...
 @Composable
-private fun TxVolumeSliderDialog(
+private fun VolumeSliderDialog(
+    title: String,
+    advice: String,
     initialValue: Int,
+    maxValue: Int,
     onChange: (Int) -> Unit,
     onDismiss: () -> Unit,
 ) {
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/settings/RadioAudioSettings.kt:836]: Valid — fixed in 6f50890. `VolumeSliderDialog` now initializes its state via a new top-level `clampVolumePercent(value, maxValue)` helper (also used by the -5/+5/drag steps), and a one-shot `LaunchedEffect` syncs the clamped value back through `onChange` when clamping changed it, so dismissing without touching the slider persists the legal value instead of re-persisting the out-of-range one. The parent `inputVolume` state also clamps on read, and the 200% bound is now the `INPUT_VOLUME_MAX` constant. Per repo convention the logic is a plain function, unit-tested in new `VolumeClampTest` (both 0..100 and 0..200 ranges, step arithmetic, and an invariant test that `INPUT_VOLUME_MAX` matches `InputAudioLevel.MAX_GAIN`).
